Product/Production/Services/DocumentQueryCore/src/main/java/gov/hhs/fha/nhinc/docquery/entity/AggregationService.java

```
@@ -229,51 +234,70 @@ protected AdhocQueryRequest setTargetHomeCommunityId(AdhocQueryRequest request,
      * @return AdhocQUery Request.
      */
     protected AdhocQueryRequest cloneRequest(AdhocQueryRequest request) {
-        AdhocQueryRequest newRequest = new AdhocQueryRequest();
-        AdhocQueryType currentQuery = request.getAdhocQuery();
-
-        AdhocQueryType newQuery = new AdhocQueryType();
-        List<SlotType1> newslotList = new ArrayList<SlotType1>();
-        List<SlotType1> slotList = currentQuery.getSlot();
-        for (SlotType1 slot : slotList) {
-            SlotType1 newSlot = new SlotType1();
-            newSlot.setName(slot.getName());
-            newSlot.setSlotType(slot.getSlotType());
-
-            ValueListType valueListType = new ValueListType();
-            List<String> valueList = new ArrayList<String>();
-            for (String value : slot.getValueList().getValue()) {
-                valueList.add(value);
+        return MessageGeneratorUtils.getInstance().clone(request);
+    }
+    
+    protected AssertionType cloneAssertionWithNewMsgId(AssertionType assertion) {
+        AssertionType newAssertion = MessageGeneratorUtils.getInstance().clone(assertion);
+        
+        newAssertion.setMessageId(new WSAHeaderHelper().generateMessageID());
+        
+        return newAssertion;
+    }
+    
+    protected Set<II> removeDuplicates(List<II> iiArray) {
```

This feels like a dangerous implementation.  It is possible (though unlikely) that external callers are relying on the default equality/hashcode behavior.  Silently substituting a subclass doesn't seem to fit what you are trying to accomplish here.  Perhaps you can have ComparableII keep the original II around as a private field, then transform the final set back into the original IIs before returning?